components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -691,31 +704,54 @@ export const useSwap = () => {
         return
       }
       setEditingFromOrToAmount('from')
-      // ToDo: Until cross-chain swaps is supported,
-      // we have this check to make sure that the toToken
+
+      if (isBridge) {
+        history.replace(
+          makeSwapOrBridgeRoute({
+            fromToken: token,
+            fromAccount: account,
+            toToken,
+            toAddress,
+            toCoin,
+            routeType: isBridge ? 'bridge' : 'swap'
```

Comment:
The `if` above already checks that this is a bridge route
```suggestion
            routeType: 'bridge'
```

Reply:
The above `if` check can still allow `swap` since `(fromAccount.accountId.coin !== toToken.coin || toCoin !== fromToken.coin)` also needs to be satisfied.

Reply:
i dont see `(fromAccount.accountId.coin !== toToken.coin || toCoin !== fromToken.coin)` in the above `if`

Reply:
Oh I was looking at https://github.com/brave/brave-core/pull/23817/files#diff-62b86bc97dc9328e9a35a7412cf52ab39018769ff344c86bfb9cb053ea085d20R621 sorry this has been fixed! https://github.com/brave/brave-core/compare/0c201842c3a57b6f0d21c6a6d94d937008b38c4c..9e3fdb52b5ff5e99a0ca5a867712a829e424a3dc